lib/CL/clEnqueueNDRangeKernel.c

```diff
@@ -241,9 +241,13 @@ POname(clEnqueueNDRangeKernel)(cl_command_queue command_queue,
   int realdev_i = pocl_cl_device_to_index(kernel->program, realdev);
   if (kernel->program->source || kernel->program->binaries[realdev_i])
     {
+#ifdef OCS_AVAILABLE
       error = pocl_llvm_generate_workgroup_function(realdev,
                                                     kernel,
                                                     local_x, local_y, local_z);
+#else
+      error = 1;
```

Comment:
If the program has been created with a binary, it won't go in this 'if'. If it does, it means that the program has not been created with binary, and as there is no compiler support, it's an error.